src/functions/indAffine.jl

```diff
@@ -26,11 +26,11 @@ immutable IndAffine{T <: RealOrComplex} <: IndicatorConvex
   end
 end
 
-IndAffine{T <: RealOrComplex}(A::AbstractArray{T,2}, b::AbstractArray{T,1}) =
-  IndAffine{T}(A, b)
+IndAffine{T <: RealOrComplex, M<:AbstractArray{T,2}, V<:AbstractArray{T,1}}(A::M, b::V) =
+  IndAffine{T,M,V}(A, b)
```

Comment:
It might be that we can. I simply followed this logic: outer constructors are to make things convenient, i.e., to allow not specifying types, to default some construction arguments, to allow for different ways of specifying the arguments (indAffine.jl is a good example of this, see the second outer construtor there); inner constructors is where all necessary checks, factorizations, normalizations occur.